src/client/dfs/dfs.c

```diff
@@ -2320,7 +2339,7 @@ dfs_read(dfs_t *dfs, dfs_obj_t *obj, d_sg_list_t *sgl, daos_off_t off,
 	if (rc)
```

Comment:
i think there was some issue with tse hanging if i don't do this. some interaction with the event API.
this wasn't from this patch though but from a different one when we needed to register 2 tasks in the read_int instead of one, so maybe i'll try what you suggested again now that it's just the read.

Reply:
this was still not working when using the array read task directly. i think once the task completes, even if you register a dependency on it in the completion cb, it returns to the user as complete.
i will have to investigate this in a separate patch, then i can make the change to avoid the extra parent task that is  needed for now.

Reply:
> this is still not working when using the array read task directly. i think once the task completes, even if you register a dependency on it in the completion cb, it returns to the user as complete.

If in child task's complete callback it register a new dependency before return, seems should not cause parent task's complete.

> i will have to investigate this in a separate patch, then i can make the change to avoid the extra parent task that is needed for now.

Maybe can refer the case in dc_obj_fetch -> obj_req_fanout (if with only one target, need not to create separate shard task). Just FYI, maybe some detailed difference in your case.